engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -233,25 +279,23 @@ object EngineJobExecutionActor {
       restarting,
       serviceRegistryActor,
       jobStoreActor,
+      dockerHashLookupActor,
+      backendName: String,
       callCachingMode)).withDispatcher(EngineDispatcher)
   }
 }
 
-sealed trait EJEAData
-case object NoData extends EJEAData
-case class EJEAJobDescriptorData(jobDescriptor: Option[BackendJobDescriptor], bjeaActorProps: Option[Props]) extends EJEAData {
-  override def toString = s"EJEAJobDescriptorData(backendJobDescriptor: ${jobDescriptor.isDefined}, bjeaActorProps: ${bjeaActorProps.isDefined})"
-}
+private[execution] sealed trait EJEAData { override def toString = getClass.getSimpleName }
 
-case class EJEAPartialCompletionData(jobResult: Option[BackendJobExecutionResponse], hashes: Option[CallCacheHashes]) extends EJEAData {
-  override def toString = s"EJEAPartialCompletionData(jobResult: ${jobResult.isDefined}, hashes: ${hashes.isDefined})"
-}
+private[execution] case object NoData extends EJEAData
+private[execution] case class JobDescriptorData(jobDescriptor: BackendJobDescriptor, bjeaActorProps: Props) extends EJEAData
 
-// Le sigh. Can't do case-to-case inheritance so let's jump through these extractor-pattern hoops...
-class EJEACompletionData(val jobResult: BackendJobExecutionResponse) extends EJEAData {
-  override def toString = s"EJEACompletionData(${jobResult.getClass.getSimpleName})"
-}
-object EJEACompletionData { def apply(jobResult: BackendJobExecutionResponse) = new EJEACompletionData(jobResult); def unapply(completionData: EJEACompletionData) = Some(completionData.jobResult) }
-case class EJEASuccessfulCompletionDataWithHashes(override val jobResult: SucceededResponse, hashes: CallCacheHashes) extends EJEACompletionData(jobResult = jobResult) {
-  override def toString = s"EJEACompletionDataWithHashes(${jobResult.getClass.getSimpleName}, ${hashes.hashes.size} hashes)"
-}
+private[execution] case object EmptyPartialCompletionData extends EJEAData
+private[execution] case class PartialCompletionDataWithSucceededResponse(response: SucceededResponse) extends EJEAData
+private[execution] case class PartialCompletionDataWithHashes(hashes: Try[CallCacheHashes]) extends EJEAData
+
+private[execution] case class CacheWriteOffCompletionData(jobResult: BackendJobExecutionResponse) extends EJEAData
+private[execution] case class CacheWriteOnCompletionData(jobResult: SucceededResponse, hashes: CallCacheHashes) extends EJEAData
+
+private[execution] case class CacheResultMatchesForHashes(hashResults: Set[HashResult], cacheResultIds: Set[MetaInfoId])
+private[execution] case class CacheResultLookupFailure(reason: Throwable)
```

Comment:
This is the pattern we've used everywhere else.

At this point I'm pretty sure all of our many "reason" boxes hold Throwables